src/workers/sourceSquadCreated.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  getContactIdByEmail,
+  addUserToContacts,
+  LIST_DRIP_CAMPAIGN,
+} from '../common';
+import { Source, SourceMember, User, UserActionType } from '../entity';
+import { SourceMemberRoles } from '../roles';
+import { insertOrIgnoreAction } from '../schema/actions';
+import { ChangeObject } from '../types';
+import { messageToJson, Worker } from './worker';
+
+interface Data {
+  source: ChangeObject<Source>;
+}
+
+const worker: Worker = {
+  subscription: 'api.source-squad-created',
+  handler: async (message, con, logger) => {
+    const data: Data = messageToJson(message);
+
+    try {
+      const { source } = data;
+      const [owner] = await con
+        .getRepository(SourceMember)
+        .createQueryBuilder('sm')
+        .select('sm."userId" as "userId"')
+        .where('sm."sourceId" = :sourceId', { sourceId: source.id })
+        .andWhere('sm."role" = :role', { role: SourceMemberRoles.Admin })
+        .orderBy('sm."createdAt"', 'ASC')
+        .limit(1)
+        .execute();
```

Comment:
Is it not possible to query this through `findOne`?

Reply:
`findOne` does not have order by and we want first admin that joined the squad which is owner.

Reply:
I could use `find` though

Reply:
tnx, added in 4715830bc6968139da3d41afd9bafafea0b614fc